client/src/components/DataFiles/DataFilesModals/DataFilesUploadModal.js

```diff
@@ -123,60 +138,25 @@ const DataFilesUploadModal = () => {
       <ModalHeader toggle={toggle}>
         Upload Files in {systemDisplayName}/{params.path}
       </ModalHeader>
-      <ModalBody>
-        <FileInputDropZone
-          onSetFiles={selectFiles}
-          onRejectedFiles={onRejectedFiles}
-          maxSize={524288000}
-          maxSizeMessage="Max File Size: 500MB"
-        />
-
+      <ModalBody styleName={containerStyleNames}>
+        <div styleName={isCompactView ? 'compact-view' : ''}>
+          <FileInputDropZone
+            onSetFiles={selectFiles}
+            onRejectedFiles={onRejectedFiles}
+            maxSize={524288000}
+            maxSizeMessage="Max File Size: 500MB"
+          />
+        </div>
         <div hidden={uploadedFiles.length === 0} style={{ marginTop: '10px' }}>
           <span style={{ fontSize: '20px' }}>
             Uploading to {systemDisplayName}/{params.path}
           </span>
-
-          <div>
-            <div
-              style={{
-                border: '1px solid black',
-                width: '100%',
-                marginTop: '5px',
-                height: '300px',
-                overflow: 'auto'
-              }}
-            >
-              <Table striped>
-                <thead>
-                  <tr>
-                    <th>Name</th>
-                    <th>Size</th>
-                    <th aria-label="null" />
-                  </tr>
-                </thead>
-                <tbody>
-                  {uploadedFiles.map((file, i) => (
-                    <tr key={file.id}>
-                      <td style={{ verticalAlign: 'middle' }}>
-                        {file.data.name}
-                      </td>
-                      <td style={{ verticalAlign: 'middle' }}>
-                        <FileLengthCell cell={{ value: file.data.size }} />
-                      </td>
-                      <td>
-                        <span className="float-right">
-                          <DataFilesUploadStatus
-                            i={file.id}
-                            removeCallback={removeFile}
-                          />
-                        </span>
-                      </td>
-                    </tr>
-                  ))}
-                </tbody>
-              </Table>
-            </div>
-          </div>
+        </div>
+        <div hidden={!showListing} styleName="dataFilesListing">
+          <DataFilesUploadModalListingTable
+            uploadedFiles={uploadedFiles}
+            setUploadedFiles={setUploadedFiles}
+          />
```

Comment:
Instead of `hidden`, could we opt to not render the component?
```suggestion
        {showListing && 
          <DataFilesUploadModalListingTable
            styleName="dataFilesListing"
            uploadedFiles={uploadedFiles}
            setUploadedFiles={setUploadedFiles}
          />
        }
```